plugin/builtin/patchtransformer/PatchTransformer_test.go

```diff
@@ -636,3 +636,68 @@ spec:
         path: /canada
 `)
 }
+
+// test for https://github.com/kubernetes-sigs/kustomize/issues/2767
+const aDeploymentResource = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: test-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - image: test-image
+        name: test-deployment
+        ports:
+        - containerPort: 8080
+          name: take-over-the-world
+          protocol: TCP
+        - containerPort: 8080
+          name: disappearing-act
+          protocol: TCP
```

Comment:
nit: Please in-line this (to L682) and mention the issue number in the comment above the test.

Reply:
no probs; i think i did that.  power just came back - coastal NC had a good storm last night.